webapp/src/ts/components/actionbar/actionbar.component.html

```diff
@@ -148,79 +169,102 @@
             </a>
-            <ul class="dropdown-menu mm-dropdown-menu with-icon" *ngIf="actionBar?.right?.relevantForms && actionBar?.right?.relevantForms?.length">
-              TODO: Order actionBar.right.relevantForms by 'title'
-              <li *ngFor="let form of actionBar.right.relevantForms; trackBy: form.code">
-                <a show-muted-modal to-state="contacts.report" to-state-params="{ id: selectedContactDoc._id, formId: form.code }">
+            <ul id="relevant-contacts-form" class="dropdown-menu mm-dropdown-menu with-icon" *ngIf="actionBar?.right?.relevantForms && actionBar?.right?.relevantForms?.length">
+              <li *ngFor="let form of actionBar.right.relevantForms; trackBy: trackByForms" [attr.id]="form.id">
+                <a (click)="actionBar.right.openContactMutedModal(form)">
                   <span [innerHTML]="form.icon | resourceIcon"></span>
-                  <span class="content">{{form.title}}</span>
+                  <span class="content">{{ form.title }}</span>
                 </a>
               </li>
             </ul>
           </span>
 
-          <span *ngFor="let group of actionBar?.right?.childTypes" mmAuth="{{group.permission}}">
-
-            <a class="mm-icon mm-icon-inverse mm-icon-caption dropdown-toggle" data-toggle="dropdown" *ngIf="group.types.length > 1">
-              <span class="fa-stack">
-                <i class="fa {{group.menu_icon}}"></i>
-                <i class="fa fa-plus fa-stack-1x"></i>
-              </span>
-              <p translate>{{group.menu_key}}</p>
-            </a>
-            <ul class="dropdown-menu mm-dropdown-menu with-icon" *ngIf="group.types.length > 1">
-            TODO order group.types by 'id' !!
-              <li *ngFor="let type of group.types; trackBy: type.id">
-                <a ui-sref="contacts.addChild({ type: type.id, parent_id: actionBarCtrl.selectedContactDoc._id })">
-                  <span [innerHTML]="type.icon | resourceIcon"></span>
-                  <span class="content" translate>{{type.create_key}}</span>
-                </a>
-              </li>
-            </ul>
+          <span *ngFor="let group of actionBar?.right?.childTypes" mmAuth="{{ group.permission }}">
+            <ng-container *ngIf="group.types?.length > 1">
+              <a class="mm-icon mm-icon-inverse mm-icon-caption dropdown-toggle" data-toggle="dropdown">
+                <span class="fa-stack">
+                  <i class="fa {{ group.menu_icon }}"></i>
+                  <i class="fa fa-plus fa-stack-1x"></i>
+                </span>
+                <p translate>{{ group.menu_key }}</p>
+              </a>
+              <ul class="dropdown-menu mm-dropdown-menu with-icon">
+                <li *ngFor="let type of group.types; trackBy: trackById">
+                  <a [routerLink]="selectedContactDoc._id ? ['/', 'contacts', selectedContactDoc._id, 'add', type.id] : ['/', 'contacts', 'add', type.id]">
```

Comment:
Is it possible we are ever in a situation where `selectedContactDoc` doesn't have an id but we still have groups? 
Aren't groups dependent on the selectedContact?

Reply:
Fixed!